File: src/main/java/com/agilesparks/rubikscube/utils/LocationInFace.java
package com.agilesparks.rubikscube.utils;

public enum LocationInFace {
    TOP('T', 0), BOTTOM('B', 1), RIGHT('R', 2), LEFT('L', 3), TOPRIGHT('E', 4), TOPLEFT('F', 5), BOTTOMRIGHT('G', 6), BOTTOMLEFT('H', 7)/*, NOTDEFINED('Z', 9)*/; //don't change this sequence, for Rubik's sake!

    private final char charValue;      // Private variable
    private final int intValue;

    LocationInFace(char charValue, int intValue) {     // Constructor
        this.charValue = charValue;
        this.intValue = intValue;
    }


    public int getIntOfChar() {              // Getter
        return charValue;
    }

    public char getChar(){ return (char) charValue;}

    public int getInt() {              // Getter
        return intValue;
    }

   
}

File: python/Production/utils/LocationInFace.py
s


    enum LocationInFace 
        TOP=0,
        BOTTOM=1,
        RIGHT=2,
        LEFT=3,
        TOPRIGHT=4,
        TOPLEFT=5,
        BOTTOMRIGHT=6,
        BOTTOMLEFT=7/*, NOTDEFINED('Z', 9)*/ //don't change this sequence, for Rubik's sake!
    
    class LocationInFaceHandle  
   
    @staticmethod
    def LocationInFace GetLocationInFace( intValue):
            switch (intValue)
            
                case 0:
                    return LocationInFace.TOP
                case 1:
                    return LocationInFace.BOTTOM
                case 2:
                    return LocationInFace.RIGHT
                case 3:
                    return LocationInFace.LEFT
                case 4:
                    return LocationInFace.TOPRIGHT
                case 5:
                    return LocationInFace.TOPLEFT
                case 6:
                    return LocationInFace.BOTTOMRIGHT
                case 7:
                    return LocationInFace.BOTTOMLEFT
                default:
                    return LocationInFace.TOP

            
           
    


   



